### PR TITLE
Improved version for thomaswue

### PR DESCRIPTION
**Pull request by thomaswue**
#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
Performance measured on Intel i9-13900K:
* Execution time: 0.92s
* Execution time of reference implementation: 120.37s

This is some tuning over the initial submission now searching for the delimiter and comparing names more than 1 byte at a time. Should be ~30% faster than initial version.

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java [hunk 1]
```diff
@@ -161,26 +174,53 @@ private static void parseLoop(long chunkStart, long chunkEnd, Result[] results,
...
+                    if ((long) nameLength >= 8) {
+                        if (UNSAFE.getLong(existingResult.nameAddress) != UNSAFE.getLong(nameAddress)) {
+                            result = false;
+                        }
+                        else {
+                            i += 8;
+                        }
+                    }
+                    else if ((long) nameLength >= 4) {
+                        if (UNSAFE.getInt(existingResult.nameAddress) != UNSAFE.getInt(nameAddress)) {
```
Comment (obourgain) [inline src/main/java/dev/morling/onebrc/CalculateAverage_thomaswue.java:198]: isn't tat line missing a `+ i ` ?

**Comment by gunnarmorling** [comment]
00:06.532, #2nd place.

> [Merge branch 'gunnarmorling:main' into main](https://github.com/gunnarmorling/1brc/pull/213/commits/789d670eff659ce374a78f85466171f12fc3483a)

@thomaswue, ah, just ran this. Did you do any actual changes besides merging main?

**Comment by gunnarmorling** [comment]
Looks like it just was that. I'm gonna go ahead and get this one in, any more improvements are welcome of course in follow-up PRs. Thanks!

**Comment by thomaswue** [comment]
Yes, I just merged main so I can try to make a GraalVM native image out of the solution of @royvanrijn ;-). On my machine this does improves execution time, but probably less on the evaluation machine given that it has less parallelism and runs for longer. And it is as simple as this:
```
native-image --gc=epsilon -O3 -march=native --enable-preview -cp target/average-1.0.0-SNAPSHOT.jar -o image_calculateaverage_royvanrijn dev.morling.onebrc.CalculateAverage_royvanrijn
```

**Pull request merged**